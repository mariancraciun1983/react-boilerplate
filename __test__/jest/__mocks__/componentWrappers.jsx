import React  from 'react';
import clone from 'clone-deep';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';

import mockStore  from './store';


const { fullStore } = mockStore;

const genericWrapper = (children) => {
  const mockStore = configureStore();
  const initialState = clone(fullStore);
  const store = mockStore(initialState);
  return (
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  )
};

export {
  genericWrapper
};
