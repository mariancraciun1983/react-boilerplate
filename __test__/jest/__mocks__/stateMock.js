export default {
  app: {
    state: "loaded"
  },
  auth: {
    authenticated: true,
    token: "abc123",
    user: {
      name: "John Doe",
      email: "john@example.com",
      avatar: "https://dummyimage.com/100x100.png/5fa2d1/ffffff"
    }
  },
  genres: {
    list: [
      {
        name: "Comedy",
        slug: "comedy"
      },
      {
        name: "Adventure",
        slug: "adventure"
      }
    ]
  },
  movies: {
    list: [
      {
        id: "0b7b8b05-6c88-4517-a969-adf89a46bc0a",
        title: "Lost Palace",
        genre: ["comedy"],
        image: "http://dummyimage.com/100x100.png/cc0000/ffffff",
        description:
          "Donec diam neque, vestibulum eget, vulputate ut, ultrices vel, augue. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec pharetra, magna vestibulum aliquet ultrices, erat tortor sollicitudin mi, sit amet lobortis sapien sapien non mi. Integer ac neque.\n\nDuis bibendum. Morbi non quam nec dui luctus rutrum. Nulla tellus.\n\nIn sagittis dui vel nisl. Duis ac nibh. Fusce lacus purus, aliquet at, feugiat non, pretium quis, lectus.\n\nSuspendisse potenti. In eleifend quam a odio. In hac habitasse platea dictumst.\n\nMaecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat.",
        price: 2.39,
        slug: "gunner-palace",
        title_lower: "lost palace"
      },
      {
        id: "f01187b5-6fe8-465b-a0b8-3764a94ae225",
        title: "Love is Eternal While It Lasts",
        genre: ["comedy", "romance"],
        image: "http://dummyimage.com/100x100.png/5fa2dd/ffffff",
        description:
          "Curabitur gravida nisi at nibh. In hac habitasse platea dictumst. Aliquam augue quam, sollicitudin vitae, consectetuer eget, rutrum at, lorem.\n\nInteger tincidunt ante vel ipsum. Praesent blandit lacinia erat. Vestibulum sed magna at nunc commodo placerat.\n\nPraesent blandit. Nam nulla. Integer pede justo, lacinia eget, tincidunt eget, tempus vel, pede.",
        price: 9.5,
        slug: "love-is-eternal-while-it-lasts",
        title_lower: "love is eternal while it lasts"
      },
      {
        id: "23ad0474-8b8d-420d-a2cc-dd54aa6ed589",
        title: "Lost World, The",
        genre: ["romance"],
        image: "http://dummyimage.com/100x100.png/cc0000/ffffff",
        description:
          "In congue. Etiam justo. Etiam pretium iaculis justo.\n\nIn hac habitasse platea dictumst. Etiam faucibus cursus urna. Ut tellus.\n\nNulla ut erat id mauris vulputate elementum. Nullam varius. Nulla facilisi.\n\nCras non velit nec nisi vulputate nonummy. Maecenas tincidunt lacus at velit. Vivamus vel nulla eget eros elementum pellentesque.\n\nQuisque porta volutpat erat. Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla. Nunc purus.",
        price: 7.4,
        slug: "lost-world-the",
        title_lower: "lost world, the"
      }
    ]
  },
  cart: {
    dict: {
      "0b7b8b05-6c88-4517-a969-adf89a46bc0a": 1,
      "23ad0474-8b8d-420d-a2cc-dd54aa6ed589": 2
    },
    saved: false
  },
  filters: {
    string: "Lo",
    string_lower: "lo"
  },
  ui: {
    language: "en",
    version: 2,
    theme: "light"
  },
  routing: {
    locationBeforeTransitions: {
      pathname: "/genre/comedy",
      search: "",
      hash: "",
      key: "sb1up4"
    }
  },
  form: {}
};
