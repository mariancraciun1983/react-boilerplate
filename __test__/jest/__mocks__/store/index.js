import clone from 'clone-deep';

import * as FullData from '../../../../__mock__/api';

const defaultStore = {
  app: {
    state: 'loading',
  },
  auth: {
    authenticated: false,
    token: null,
    user: null,
  },
  cart: {
    dict: {},
    saved: true,
  },
  filters: {
    string: '',
    string_lower: '',
  },
  genres: {
    list: null,
  },
  movies: {
    list: null,
  },
  ui: {
    language: 'en',
    version: 0,
    theme: 'light',
  },
  routing: {},
  form: {},
};

const fullStore = clone(defaultStore);
fullStore.state = 'loaded';
fullStore.genres.list = FullData.genres;
fullStore.movies.list = FullData.movies.slice(0,200);
export default {
  defaultStore,
  fullStore,
}
