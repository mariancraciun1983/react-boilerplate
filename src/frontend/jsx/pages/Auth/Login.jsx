import React from 'react';
import PropTypes from 'prop-types';
import { SubmissionError } from 'redux-form';
import { push } from 'react-router-redux';
import { connect } from 'react-redux';
import * as Api from '../../../core/middleware/api';
import { authIsAuthenticated } from '../../../core/selectors';
import { authSet } from '../../../core/stores/reducers/auth';
import LoginForm from './components/LoginForm';

const stateToProps = (state) => ({
  isAuthenticated: authIsAuthenticated(state),
});

const actionsToProps = (dispatch) => ({
  redirect: (to) => dispatch(push(to)),
  authDataSet: (data) => dispatch(authSet(data)),
});

class Login extends React.PureComponent {
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidMount() {
    const { redirect, isAuthenticated } = this.props;
    if (isAuthenticated) {
      redirect('/');
    }
  }

  handleSubmit(values) {
    const { authDataSet, redirect } = this.props;
    return new Promise((resolve, reject) => {
      const success = (data) => {
        if (data && data.token && data.user) {
          const { token, user } = data;
          authDataSet({ token, user });
          redirect('/');
          resolve();
        } else {
          reject(new SubmissionError({ _error: 'Failed to process the form' }));
        }
      };
      const error = () => {
        reject(new SubmissionError({ _error: 'Something went wrong' }));
      };
      Api.authLogin({ ...values })
        .then(success)
        .catch(error);
    });
  }

  render() {
    const initialValues = { email: 'john@example.com', password: 'password' };
    return (
      <>
        <div className="container">
          <div className="row">
            <div className="col col-6 offset-3">
              <div className="card-header">
                <h1>Login</h1>
              </div>
              <div className="card-body">
                <LoginForm onSubmit={this.handleSubmit} initialValues={initialValues} />
              </div>
            </div>
          </div>
        </div>
      </>
    );
  }
}


Login.propTypes = {
  redirect: PropTypes.func.isRequired,
  authDataSet: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
};


export const Component = Login;
export default connect(
  stateToProps,
  actionsToProps,
)(Login);
