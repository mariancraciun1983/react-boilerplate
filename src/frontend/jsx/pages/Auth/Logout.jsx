import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';

import { authLogout } from '../../../core/stores/reducers/auth';

const actionsToProps = (dispatch) => ({
  authUserLogout: () => dispatch(authLogout()),
  redirect: (to) => dispatch(push(to)),
});

class Logout extends React.PureComponent {
  componentDidMount() {
    const { authUserLogout, redirect } = this.props;
    authUserLogout();
    setTimeout(() => {
      redirect('/');
    }, 300);
  }

  render() {
    return (
      <div style={{ minHeight: 300 }}>
        <h4>Logout</h4>
        <p>
          You are being logged out. Please wait while we are redirecting you to the homepage
        </p>
      </div>
    );
  }
}

Logout.propTypes = {
  redirect: PropTypes.func.isRequired,
  authUserLogout: PropTypes.func.isRequired,
};

export const Component = Logout;
export default connect(
  null,
  actionsToProps,
)(Logout);
