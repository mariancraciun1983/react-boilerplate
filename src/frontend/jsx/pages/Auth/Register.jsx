import React from 'react';
import PropTypes from 'prop-types';
import { SubmissionError } from 'redux-form';
import { push } from 'react-router-redux';
import { connect } from 'react-redux';
import * as Api from '../../../core/middleware/api';
import { authIsAuthenticated } from '../../../core/selectors';
import { authSet } from '../../../core/stores/reducers/auth';
import RegisterForm from './components/RegisterForm';

const stateToProps = (state) => ({
  isAuthenticated: authIsAuthenticated(state),
});

const actionsToProps = (dispatch) => ({
  redirect: (to) => dispatch(push(to)),
  authUserSet: (data) => dispatch(authSet(data)),
});

class Register extends React.PureComponent {
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidMount() {
    const { isAuthenticated, redirect } = this.props;
    if (isAuthenticated) {
      redirect('/');
    }
  }

  handleSubmit(values) {
    const { authUserSet, redirect } = this.props;
    return new Promise((resolve, reject) => {
      const success = (data) => {
        if (data && data.token && data.user) {
          const { token, user } = data;
          authUserSet({ token, user });
          redirect('/');
          resolve();
        } else {
          reject(new SubmissionError({ _error: 'Failed to process the form' }));
        }
      };
      const error = () => {
        reject(new SubmissionError({ _error: 'Something went wrong' }));
      };
      Api.authRegister({ ...values })
        .then(success)
        .catch(error);
    });
  }

  render() {
    const initialValues = { name: 'John Doe', email: 'john@example.com', password: 'password' };
    return (
      <>
        <div className="container">
          <div className="row">
            <div className="col col-6 offset-3">
              <div className="card-header">
                <h1>Register</h1>
              </div>
              <div className="card-body">
                <RegisterForm onSubmit={this.handleSubmit} initialValues={initialValues} />
              </div>
            </div>
          </div>
        </div>
      </>
    );
  }
}

Register.propTypes = {
  redirect: PropTypes.func.isRequired,
  authUserSet: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
};

export const Component = Register;
export default connect(
  stateToProps,
  actionsToProps,
)(Register);
