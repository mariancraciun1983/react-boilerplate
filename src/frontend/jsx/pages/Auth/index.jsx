import React from 'react';
import { Switch, Route, Redirect } from 'react-router';

import Login from './Login';
import Logout from './Logout';
import Register from './Register';

const routes = () => (
  <Switch>
    <Route component={Login} path="/auth/login" exact />
    <Route component={Register} path="/auth/register" exact />
    <Route component={Logout} path="/auth/logout" exact />
    <Redirect to={{ state: { error404: true, sourceModule: 'auth' } }} />
  </Switch>
);

export default { routes };
