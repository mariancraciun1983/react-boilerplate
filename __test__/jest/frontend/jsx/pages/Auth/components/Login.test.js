/// <reference types="jest" />
import React from 'react';
import renderer from 'react-test-renderer';

import {Component} from '../../../../../../../src/frontend/jsx/pages/Auth/Login';
import { genericWrapper } from '../../../../../__mocks__/componentWrappers';


describe("components/pages/Auth/Login", () => {
  let props;
  beforeEach(() => {
    props =  {
      isAuthenticated: false,
      authDataSet: jest.fn(),
      redirect: jest.fn(),
    }
  });

  it("Renders the Login", () => {
    //it's not trivial to test, so we will just test it how well it renders
    expect.assertions(1);
    const tree = renderer
    .create(genericWrapper(<Component {...props} />))
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
