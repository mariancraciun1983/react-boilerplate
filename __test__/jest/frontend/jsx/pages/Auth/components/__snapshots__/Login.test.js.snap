// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/Auth/Login Renders the Login 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col col-6 offset-3"
    >
      <div
        className="card-header"
      >
        <h1>
          Login
        </h1>
      </div>
      <div
        className="card-body"
      >
        <div>
          <form
            onSubmit={[Function]}
          >
            <div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="email"
                >
                  Email
                </label>
                <input
                  autoComplete="email"
                  className="form-control"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Email"
                  type="email"
                  value=""
                />
              </div>
            </div>
            <div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="password"
                >
                  Password
                </label>
                <input
                  autoComplete="current-password"
                  className="form-control"
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div
              className=" "
            >
              <button
                className="btn btn-success btn-lg"
                disabled={false}
                type="submit"
              >
                Login
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
